pkg/app/piped/apistore/eventstore/store.go

```diff
@@ -116,19 +123,37 @@ func (s *store) sync(ctx context.Context) error {
 		return nil
 	}
 
-	// Eliminate events that have duplicated key.
-	filtered := make(map[string]*model.Event, len(resp.Events))
+	// Group the events by key, sorted by newest.
+	eventsByKey := make(map[string][]*model.Event, len(resp.Events))
 	for _, e := range resp.Events {
-		filtered[e.EventKey] = e
+		eventsByKey[e.EventKey] = append(eventsByKey[e.EventKey], e)
+	}
+	// Ensure that the 10 least recent entries get outdated.
+	// NOTE: We don't support batch update, that's why we have a constant number of updates to make.
+	outdated := make([]*pipedservice.ReportEventStatusesRequest_Event, 0)
+	for _, es := range eventsByKey {
+		for i := 1; i < len(es) && i <= numToMakeOutdated; i++ {
+			if es[i].Status != model.EventStatus_EVENT_NOT_HANDLED {
+				continue
+			}
+			outdated = append(outdated, &pipedservice.ReportEventStatusesRequest_Event{
+				Id:                es[i].Id,
+				Status:            model.EventStatus_EVENT_OUTDATED,
+				StatusDescription: fmt.Sprintf("The new event %q has been created", es[0].Id),
+			})
+		}
 	}
+	if len(outdated) > 0 {
+		if _, err := s.apiClient.ReportEventStatuses(ctx, &pipedservice.ReportEventStatusesRequest{Events: outdated}); err != nil {
+			return fmt.Errorf("failed to report event statuses: %w", err)
+		}
+		s.logger.Info(fmt.Sprintf("successfully made %d events OUTDATED", len(outdated)))
+	}
+
```

Comment:
I think updating the event status is not the responsibility of the `event` store.
The `event` store does only one thing is to keep its local data to be synced with the control plane.
So that logic should belong to the `event-watcher` component.

Reply:
I feel the same way 👍